[docs] appearance, formatting and wording tweaks/fixes (#29567)
expo/expo: docs/pages/build-reference/caching.mdx

## JavaScript dependencies

> EAS Build runs a npm cache server that can speed up downloading JavaScript dependencies for your build jobs. Projects that are using npm or yarn v2 will use the cache by default, but yarn v1 will require that you apply this [workaround](/build-reference/npm-cache-with-yarn).

Reviewer: ```suggestion
EAS Build runs a npm cache server that can speed up downloading JavaScript dependencies for your build jobs. Projects using npm or Yarn Modern (v2+) will use the cache by default. However, Yarn Classic (v1) will require that you apply [this workaround](/build-reference/npm-cache-with-yarn).
```

See this guideline (https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#follow-external-product-casing) in writing style guide for more info on preferring Yarn vs yarn.
